gfs/benchmark/performance_test.go

```diff
@@ -0,0 +1,190 @@
+package benchmark
+
+import (
+	"bytes"
+	"gfs/chunkserver"
+	"gfs/chunkserver/services"
+	"gfs/client"
+	"gfs/master"
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// ----------- BENCHMARK CONFIGURATIONS HERE --------------------------
+
+var masterServerPort = ":9000"
+var chunkServerPortBase = 10000
+var NUM_CHUNK_SERVERS = 3
+
+var NTRIALS = 1 // repeat test NTRIALS time to collect
+
+var ARCHITECTURE = "ASYNC"           // SEQ for sequential; ASYNC for async
+var INPUT_FILE = "textfile_500b.txt" // input files vary in size
+var INPUT_FILES = []string{
+	"textfile_500b.txt",
+	"textfile_1kb.txt",
+	"textfile_1.5kb.txt",
+	"textfile_5kb.txt",
+	"textfile_10kb.txt",
+	"textfile_20kb.txt",
+	"textfile_30kb.txt",
+	"textfile_40kb.txt",
+	"textfile_64kb.txt",
+	"textfile_100kb.txt",
+	"textfile_250kb.txt",
+	"textfile_500kb.txt",
+	"textfile_750kb.txt",
+
+	// x1000 larger in magnitude...
+	"textfile_1mb.txt",
+	"textfile_2mb.txt",
+	"textfile_3mb.txt",
+	"textfile_5mb.txt",
+
+	"textfile_6mb.txt",
+	"textfile_9mb.txt",
+	"textfile_10mb.txt",
+	"textfile_15mb.txt",
+
+	"textfile_18mb.txt",
+	"textfile_20mb.txt",
+
+	// any larger, we are approaching the uppbound of GRPC's message capacity.
+	// ERROR with GRPC
+}
```

Comment:
Rather than hard-code, we I tried to use `ioutils.ReadDir()`to get file names, but I got some gRPC error involving UTF-8 encoding.